scanpy/tests/test_highly_variable_genes.py

```diff
@@ -338,23 +338,26 @@ def test_pearson_residuals_batch(pbmc3k_parametrized_small, subset, n_top_genes)
         ),
     ],
 )
-def test_compare_to_upstream(
+@pytest.mark.parametrize("array_type", ARRAY_TYPES)
+def test_compare_to_upstream(  # noqa: PLR0917
     request: pytest.FixtureRequest,
     func: Literal["hvg", "fgd"],
     flavor: Literal["seurat", "cell_ranger"],
     params: dict[str, float | int],
     ref_path: Path,
+    array_type: list,
 ):
     if func == "fgd" and flavor == "cell_ranger":
         msg = "The deprecated filter_genes_dispersion behaves differently with cell_ranger"
         request.node.add_marker(pytest.mark.xfail(reason=msg))
     hvg_info = pd.read_csv(ref_path)
-
     pbmc = pbmc68k_reduced()
     pbmc.X = pbmc.raw.X
-    pbmc.var_names_make_unique()
 
+    pbmc.var_names_make_unique()
     sc.pp.normalize_per_cell(pbmc, counts_per_cell_after=1e4)
+    pbmc.X = array_type(pbmc.X)
```

Comment:
`normalize_per_cell` propagates too many differences and throws off everything downstream of this.

Reply:
Yes!